Source/common/SNTFileInfo.m

```diff
@@ -572,6 +572,11 @@ - (NSDictionary *)embeddedPlist {
     NSData *cmdData = [self safeSubdataWithRange:NSMakeRange(offset, sz_segment)];
     if (!cmdData) return nil;
 
+    struct load_command *lc = (struct load_command *)[cmdData bytes];
+    if (lc->cmdsize < sizeof(struct load_command)) {
```

Comment:
As `lc` isn't needed again you could combine these (which also avoids having `lc` re-declared in an inner scope of a potentially different type):

```
if (((struct load_command *lc)[cmdData bytes])->cmdsize < sz_segment) return nil;
```

This also checks the size is at least `sz_segment` size.

Reply:
Consolidated, however some valid commands are smaller than `segment_command` so left the comparison against `sizeof(struct load_command)` (which is just the common fields between all command types).